### PR TITLE
implement the QP solver

### PR DESCRIPTION
**Pull request by lukeli1990**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -9,11 +9,14 @@ CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
...
 FileIO = "5789e2e9-d7fb-5bc7-8068-2c6fae9b9549"
+Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
+JuMP = "4076af6c-e467-56ae-b986-b466b2749572"
 KernelAbstractions = "63c18a36-062a-441e-b654-da1e3ab1ce7c"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MPI = "da04e1cc-30fd-572f-bb4f-1f8673147195"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
```
Comment (kibaekkim) [inline Project.toml:20]: We need to clean up `Project.toml` before merge.

**Review by kibaekkim — COMMENTED** [review]
@lukeli1990 Can you take a closer look at all the codes and clean them up? For example, should we remove `old_structure` and `src/other_test`? And also remove `test/bowen_temp.jl`, `case9.m`, `case118.m`, and `case1354pegase.m`?

**Comment by michel2323** [comment]
There is some error due to the artifact. What is the issue with `case9_onehour_60.Pd`. Do you want me to add that file to the ExaData artifact or is something else wrong?

**Comment by lukeli1990** [comment]
@michel2323 @kibaekkim Hi Michel, I didn't touch the artifact file, not on purpose. But I guess when I linked the packages (SQP.jl, SQOPF.jl and ExaADMM.jl), I might introduced errors to the .toml files unintentionally.

**Comment by michel2323** [comment]
Merged https://github.com/exanauts/ExaAdmm.jl/pull/43 with a merge back and rebase.

**Pull request closed without merging**